scripts/Tools/xmlconvertors/convert-grid-v1-to-v2

```diff
@@ -0,0 +1,217 @@
+
+        domain_data.append( (name, nx, ny, file_masks, path_masks, desc) )
+
+    v2domains = v2file_obj.make_child("domains")
+
+    for name, nx, ny, file_masks, path_masks, desc in domain_data:
+        attribs = {"name":name} if name else {}
+        domain_block = v2file_obj.make_child("domain", attributes=attribs, root=v2domains)
+
+        v2file_obj.make_child("nx", root=domain_block, text=nx)
+        v2file_obj.make_child("ny", root=domain_block, text=ny)
+
+        file_to_attrib = OrderedDict()
+        for component, mask_values in file_masks.iteritems():
+            for mask_value, filename in mask_values.iteritems():
+                if filename is None:
+                    continue
+
+                try:
+                    path = path_masks[component][mask_value]
+                except KeyError:
+                    path = "$DIN_LOC_ROOT/share/domains"
+
+                fullfile = os.path.join(path, filename)
+                mask_value = mask_value if mask_value not in ["reg", name] else ""
+                file_to_attrib.setdefault(fullfile, OrderedDict()).setdefault(mask_value, []).append(component)
+
+        for filename, masks in file_to_attrib.iteritems():
+            attrib = {}
+            expect(len(masks) == 1, "Bad mask")
+            for mask, components in masks.iteritems():
+                attrib["grid"] = "|".join(components)
+
+            if mask:
+                attrib["mask"] = mask
+
+            v2file_obj.make_child("file", attributes=attrib, root=domain_block, text=filename)
+
+        if desc:
+            v2file_obj.make_child("desc", root=domain_block, text=desc)
+
+###############################################################################
+def convert_grids(v1file_obj, v2file_obj):
+###############################################################################
+    grid_data = [] # (compset, lname, sname, alias, support)
+
+    v1grids = v1file_obj.get_child(name="grids")
+    v1grid = v1file_obj.get_children(name="grid", root=v1grids)
+    for grid_block in v1grid:
+        attrib = v1file_obj.attrib(grid_block)
+
+        compset = attrib["compset"] if "compset" in attrib else None
+        expect(attrib.keys() in [ ["compset"], [] ],
+               "Unexpected attribs: {}".format(attrib))
+
+        lname   = v1file_obj.get_element_text("lname", root=grid_block)
+        sname   = v1file_obj.get_element_text("sname", root=grid_block)
+        alias   = v1file_obj.get_element_text("alias", root=grid_block)
+        support = v1file_obj.get_element_text("support", root=grid_block)
+
+        for child in v1file_obj.get_children(root=grid_block):
+            expect(v1file_obj.name(child) in ["lname", "sname", "alias", "support"],
+                   "Unhandled child of grid '{}'".format(v1file_obj.name(child)))
+
+        grid_data.append((compset, lname, sname, alias, support))
+
+    v2grids = v2file_obj.make_child("grids")
+
+    # TODO: How to leverage model_grid_defaults
```

Comment:
I had no idea how to use model_grid_defaults in this conversion.

Reply:
These are features that were not in v1.  model_grid_defaults is the grid that component will use if it is not explicitly called out in a grid definition.   So CISM2 will always use gland4 and rtm will always use r05 unless they are explicit in the grid name.  You may want to add some defaults for your mpas based grids.